### DISCUSSION TITLE
How do you use fs.exists()?

### DISCUSSION THREAD

**Discussion by ghost** [tauri-apps/tauri#5530] [Q&A] [answered]
Why fs.exists() returns void and not boolean?

I want to do something like this
```ts
if (await exists("FILE_NAME", { dir: BaseDirectory.App })) {
  console.log('file exists')
} else {
  console.log('no such file')
}
```

# SOLUTION:
```ts
if (await exists("FILE_NAME", { dir: BaseDirectory.App }) as unknown as boolean) {
  console.log('file exists')
} else {
  console.log('no such file')
}
```

**Comment by FabianLars** [comment, marked as answer]
If you use typescript you can do `await exists() as unknown as boolean`, if you don't use typescript just do what you're doing already. It _does_ return a boolean, only the type annotation is wrong.

**Comment by ghost** [comment]
I found some comments. It seems to be wrong type annotations. It actually returns boolean

![image](https://user-images.githubusercontent.com/77569421/199546775-05b8d5c3-de15-4e91-9bdb-e7027911e122.png)

**Reply by FabianLars** [reply]
Yeah i have to explain that daily :D Hopefully we can release 1.2 soon...

**Comment by ghost** [comment]
Yes, thank you!
